### PR TITLE
Add softmax activation for moe_fused_gate

### PR DESCRIPTION
**Pull request by gaopengff**
## Motivations
This kernel is used by **biased_grouped_topk** with activation of **sigmoid**. However, the method of **grouped_topk** in sglang used **softmax** and does not have bias. In that case it could reuse this moe_fused_gate kernels with some modifications.
Sglang PR https://github.com/sgl-project/sglang/pull/31126 will enable **biased_grouped_topk** and **grouped_topk** with this kernel.

## Changes
1. Added softmax activation for moe_fused_gate kernel and corresponding ut cases.
2. Removed **torch.compile** for torch native method in ut as eager mode actually makes ut faster.

### CODE REVIEW

File: tests/test_moe_fused_gate.py [hunk 1]
```diff
@@ -98,19 +104,28 @@ def biased_grouped_topk_native(
...
 
     torch.manual_seed(seq_length)
     tensor = torch.rand((seq_length, num_experts), dtype=dtype, device="xpu")
     scores = tensor.clone()
-    bias = torch.rand(num_experts, dtype=dtype, device="xpu")
+    if scoring_func == "1":
+        # grouped_topk with softmax activation doesn't have bias
+        bias = torch.zeros(num_experts, dtype=dtype, device="xpu")
+    else:
+        bias = torch.rand(num_experts, dtype=dtype, device="xpu")
```
[the remark this answers is not in the document]
Comment (gaopengff) [reply tests/test_moe_fused_gate.py]: Fixed.

File: python/sgl_kernel/moe.py [hunk 2]
```diff
@@ -112,13 +118,18 @@ def moe_fused_gate(
     # routed_scaling_factor: if > 0, the experts will be scaled by this factor
     # apply_routed_scaling_factor_on_output: if true, output will be
     #   scaled by the routed_scaling_factor
+    scoring_func_int = _SCORING_FUNC_MAP.get(scoring_func.lower())
+    assert (
+        scoring_func_int is not None
+    ), f"Unknown scoring_func '{scoring_func}', must be one of {list(_SCORING_FUNC_MAP.keys())}"
```
[the remark this answers is not in the document]
Comment (gaopengff) [reply python/sgl_kernel/moe.py]: Fixed.

File: src/torch_extension_sycl.cc [hunk 3]
```diff
@@ -62,7 +62,8 @@ TORCH_LIBRARY_FRAGMENT(sgl_kernel, m) {
   m.impl("swiglu_gpt_oss_sigmoid_alpha", torch::kXPU, &swiglu_gpt_oss_sigmoid_alpha);
   m.def(
       "moe_fused_gate(Tensor input, Tensor bias, int num_expert_group, int topk_group, int topk, int "
-      "num_fused_shared_experts, float routed_scaling_factor, bool apply_routed_scaling_factor_on_output) -> "
+      "num_fused_shared_experts, int scoring_func, float routed_scaling_factor, bool "
```
[the remark this answers is not in the document]
Comment (gaopengff) [reply src/torch_extension_sycl.cc]: Not a problem since there is no backward operation here.

File: python/sgl_kernel/moe.py [hunk 4]
```diff
@@ -4,6 +4,11 @@
 
 from .utils import is_xe2_arch
 
+_SCORING_FUNC_MAP = {
```
Comment (airMeng) [inline python/sgl_kernel/moe.py]:
```suggestion
_MOE_SCORING_FUNC_MAP = {
```

**Review by airMeng — APPROVED** [review]

File: python/sgl_kernel/moe.py (hunk 4 above)
Comment (gaopengff) [reply python/sgl_kernel/moe.py]: Done.
